tools/plugins/com.liferay.ide.server.tomcat.ui/src/com/liferay/ide/server/tomcat/ui/editor/LiferayServerSettingsEditorSection.java

```diff
@@ -962,6 +963,20 @@ else if (path.equals(installDirPath))
 
 			}
 			*/
+            String memoryValue = memoryArgs.getText();
+            String[] memory = DebugPlugin.parseArguments( memoryValue );
+            if( !CoreUtil.isNullOrEmpty( memoryValue ) )
+            {
+                for( String str : memory )
+                {
+                    if( !( str.startsWith( "-X" ) || //$NON-NLS-1$
+                        str.startsWith( "-XX:PermSize" ) || str.startsWith( "-XX:MaxPermSize" ) ) )//$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
if you have already checked -X there is no need to check for -XX:PermSize, etc because it would have been covered by the previous check.